api/nodes/views.py

```diff
@@ -1282,6 +1282,8 @@ class NodeFilesList(generics.ListAPIView, WaterButlerMixin, ListFilterMixin, Nod
         base_permissions.TokenHasScope,
     )
 
+    ordering = ('materialized_path',)  # default ordering
```

Comment:
In this occasion, it's an instance of OsfStorageFile which doesn't have a date_modified property/field. It appears that everything inherits from FileNode and FileNode has a stored_object property which has a last_touched property. For OsfStorageFile this appears to be None regardless of me having touched it. GitHubFile does, however, have values upon creation. I could test the other types to determine if last_touched values can be counted on.

OsfStorageFile does have a history property at it's top level which has entries with a modified value that is a unicode string of a naive datetime with second precision. 

I could add a property to FileNode to act as an interface over the different modified options:
```python
@property
def date_modified(self):
    try:
        if isinstance(self, OsfStorageFile):
            return datetime.strptime(self.history[-1]['modified'], "magic")
        else:
            return self.stored_object.last_touched
    except Exception as ex:
        return None
```

But I'm not sure this doesn't go outside of the scope of the fix. `s/date_modified/id/g` would fix the bug without adding complexity.